UI: Fix layout height

cdedreuille opened the pull request:
Closes https://github.com/storybookjs/storybook/issues/24261

## What I did

To support older browsers we can't use `100dvh` to set the height of the layout to be always visible on mobile. Instead, the most reliable way is to set the height using javascript. A common pattern is to set a css variable for `--vh` that you can then use elsewhere if needed it on your app.

## Checklist for Contributors

### Testing

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to communicate how to test your changes -->

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

<!-- Please include the steps to test your changes here. For example:

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Open Storybook in your browser
3. Access X story

-->

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

cdedreuille committed:
Update Layout.tsx

cdedreuille committed:
Add a default value before render

ndelangen commented on code/ui/manager/src/components/layout/Layout.tsx:
```diff
@@ -171,9 +171,22 @@ export const Layout = ({ managerLayoutState, setManagerLayoutState, ...slots }:
     return {
       width: '100%',
-      height: '100svh', // We are using svh to use the minimum space on mobile
+      height: 'var(--vh, 100vh)',
```
Emotion supports array values for fallbacks:
https://emotion.sh/docs/object-styles#fallbacks

```suggestion
      height: ['100dvh', '100vh],
```

IIUC, the bug that requires `100dvh` only happens on `iOS` and that supports the CSS feature?

JReinhold replied:
This is more about supporting old browser versions, and we haven't yet decided which versions to support in SB 8.0. https://caniuse.com/viewport-unit-variants

The fallback syntax looks interesting, I can't find any info on how this works? Is it a browser feature, or does Emotion do something clever with a list of CSS features per user agent?

based on the docs you linked, I think the syntax is the other way around in your suggestion.

cdedreuille replied:
Oh that's interesting. I didn't know Emotion had fallbacks. I'll try it out now.

ndelangen replied:
Having the same property twice on a single class is a CSS feature.
CSS is fault-tolerant.

Emotion allows you to use this feature with the array syntax.

True, the exact implementation might have to be reversed from the code suggestion I added.

cdedreuille committed:
Use array instead with a fallback

cdedreuille commented:
@JReinhold @ndelangen I've implemented the change and removed the JS code 😊 It does work as intended.

![CleanShot 2023-10-06 at 16 42 38@2x](https://github.com/storybookjs/storybook/assets/1540635/3fa11aea-f208-421f-b85e-80f1b84a8a0c)

cdedreuille committed:
Update Layout.tsx
